src/structures/interfaces/TextBasedChannel.js

```diff
@@ -77,8 +77,11 @@ class TextBasedChannel {
    */
 
   /**
-   * Types of mentions to enable in MessageMentionOptions
-   * @typedef {'role' | 'users' | 'everyone'} MessageMentionTypes
+   * Types of mentions to enable in MessageMentionOptions. It can be one of the following strings:
+   * - `role`
+   * - `users`
+   * - `everyone`
+   * @typedef {string} MessageMentionTypes
```

Comment:
```suggestion
   * @typedef {string[]} MessageMentionTypes
```
Thanks for clarifying - in that case this should be typed as an array of strings - I realise now that a single literal type was never correct either way.

You might also want to change line 62 while you're here, which also has a string literal type.

Reply:
I'm not too sure about this change, I'm looking at the other `*Type`'s and none of them has an array as their type; when using it as an array is applicable (e.g. [ClientOptions#partials](https://discord.js.org/#/docs/main/master/typedef/ClientOptions)) it states it as an array there, not into the actual type. This is just my opinion however, let me know what you think. I'm just worried about consistency.

For the line you referred to, should I create a new typedef for that option or simply specify it as `string` and state the possible values in the description?

Reply:
Ahh, my mistake - this should be string, yeah.

Not sure on the other one